### ISSUE TITLE
Cannot open any file using `open with` on mac

### ISSUE THREAD

**Issue by Doublonmousse** [xournalpp/xournalpp#5229]
### Operating System

Mac OS

### (Linux Only) Display Server

None

### Installation Method

Official release

### Xournal++ Version

1.2.1

### libgtk Version

3.24.38

### Bug Description

On a mac (using Sonoma with a M2 processor), even after giving full disk access to xournal++, files cannot be opened using finder. They can only be opened when navigating to the file using the `open file` menu.

Opening any xopp file will result in this error 
<img width="395" alt="image" src="https://github.com/xournalpp/xournalpp/assets/115779707/799be70b-311d-498c-875b-cfb4a397cf30">

### Expected Behaviour

The file is opened and no spurrious error of the type "incompatible file type" should appear.

### Steps to Reproduce

1. Have a valid `.xopp` file
2. Open the file with `xournal++` from finder
3. See an error message

**Comment by rolandlo** [comment]
Opening files from finder hasn't been implemented yet. We are aware of this problem, see https://github.com/xournalpp/xournalpp/pull/3492#issuecomment-950113675 and https://github.com/xournalpp/xournalpp/pull/3492#issuecomment-950222584, but thanks for filing an issue on the issue tracker.

**Comment by Doublonmousse** [comment]
Yes, I could not find any related issue so I opened one.

If this is of any interest, I know that `rnote` has finder integration and is based on gtk (although `gtk4` not `gtk3`)

**Comment by rolandlo** [comment]
Thanks for the hint. Actually I think that already having migrated to `GApplication` we can just add the `G_APPLICATION_HANDLES_OPEN` flag to 
https://github.com/xournalpp/xournalpp/blob/ce8039980d452cd2f1a47fb7970e4e1c28491f98/src/core/control/XournalMain.cpp#L53-L54 and implement
https://github.com/xournalpp/xournalpp/blob/ce8039980d452cd2f1a47fb7970e4e1c28491f98/src/core/control/XournalMain.cpp#L425-L430
Compare [this example](https://gitlab.gnome.org/GNOME/glib/-/blob/HEAD/gio/tests/gapplication-example-open.c).

**Comment by rolandlo** [comment]
I tried to fix this.
- Added a `CFBundleDocumentTypes` section into `Info.plist`, such that `xopp`-files are associated with Xournal++ (works in finder).
- Added the `G_APPLICATION_HANDLES_OPEN` flag in `XournalMain.cpp`
- Removed the `GOptionEntry` for `G_OPTION_REMAINING`, so that the remaining arguments are considered as names of file(s) to be opened. Now running Xournal++ via 
```term 
jhbuild run xournalpp
``` 
or 
```term
open -a Xournal++.app
``` 
will use the activate signal handler (after the startup signal handler), while running Xournal++ with a file via 
```term
jhbuild run xournalpp doc.xopp
``` 
or 
```term
open -a Xournal++.app --args doc.xopp
```
will use the open signal handler (also after the startup signal handler).
Unfortunately for Xournal++.app there is a difference between 
```term
open -a Xournal++.app --args doc.xopp
``` 
and 
```term
open -a Xournal++.app doc.xopp
``` 
(or simply `open doc.xopp`, which seems to be what Finder uses), 
whereas for RNote those work the same. I haven't discovered the cause yet. When running `open doc.xopp` Xournal++ is still opened, but I get an error message
```term
The document "doc.xopp" could not be opened. Xournal++ cannot open files in the "Xournal++ Document" format.
```

**Comment by WailRoth** [comment]
Hi! Still not fixed on V1.2.3

**Comment by Timtonix** [comment]
Not fixed today...

**Comment by rolandlo** [comment]
> It is, but the fix will only appear in v1.3.0. @rolandlo was there a reason why #5256 was targetted at master and not at the release branch? Could it be backported easily?

I can't see a reason why it wasn't targetted at release-1.2. Probably I just forgot about this possibility. I'll try to backport it tomorrow.
